share/build/buildlib.spio

```diff
@@ -86,80 +92,96 @@ def buildlib(bldroot, installpath, case):
+    if debug:
+        cmake_opts += "-DPIO_ENABLE_LOGGING=ON "
+    # Case changes for NetCDF/NETCDF forces us to do this. For other packages
+    # what's already in the env should work
+    if "NETCDF_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_PATH:PATH={os.environ['NETCDF_PATH']} "
+    if "NETCDF_C_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_C_PATH:PATH={os.environ['NETCDF_C_PATH']} "
+    if "NETCDF_FORTRAN_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_Fortran_PATH:PATH={os.environ['NETCDF_FORTRAN_PATH']} "
+
+    # scorpio needs a little help finding hdf5. The FindHDF5 in scopio uses the
+    # env var "HDF5" instead of "HDF5_ROOT". If HDF5_ROOT is not set, but we have
+    # h5dump in our path, we can use that info to set up HDF5. For now, we will
+    # assume only HDF5_ROOT is what matters.
+    which_h5dump = shutil.which('h5dump')
+    if "HDF5_ROOT" in os.environ:
+        os.environ["HDF5"] = os.environ["HDF5_ROOT"]
+    # elif which_h5dump is not None:
+    #     os.environ["HDF5"] = os.path.dirname(os.path.dirname(which_h5dump))
+
+    if "HDF5" in os.environ:
+        cmake_opts += "-DWITH_HDF5=On "
+
+    # Same deal with libz and szip
+    if "ZLIB_ROOT" in os.environ:
+        os.environ["LIBZ"] = os.environ["ZLIB_ROOT"] # LIBZ/ZLIB
+    if "SZIP_ROOT" in os.environ:
+        os.environ["SZIP"] = os.environ["SZIP_ROOT"]
+
+    if "PNETCDF_PATH" in os.environ:
+        cmake_opts += f"-DPnetCDF_PATH:PATH={os.environ['PNETCDF_PATH']} "
     else:
-        cmake_opts = '"-D GENF90_PATH=' + srcroot + '/libraries/parallelio/scripts/ "'
-
-    stdargs = get_standard_makefile_args(case, shared_lib=True)
-
-    gmake_vars = (
-        "CASEROOT={caseroot} COMP_NAME={pio_model} "
-        "USER_CMAKE_OPTS={cmake_opts} "
-        "PIO_LIBDIR={pio_dir} CASETOOLS={casetools} "
-        "USER_CPPDEFS=-DTIMING".format(
-            caseroot=caseroot,
-            pio_model=pio_model,
-            cmake_opts=cmake_opts,
-            pio_dir=pio_dir,
-            casetools=casetools,
-        )
-    )
-
-    if scorpio_src_root_dir is not None:
-        gmake_vars += (
-            " IO_LIB_SRCROOT={scorpio_src_root_dir} "
-            " IO_LIB_v1_SRCDIR={scorpio_classic_dir} "
-            " IO_LIB_v2_SRCDIR={scorpio_dir} ".format(
-                scorpio_src_root_dir=scorpio_src_root_dir,
-                scorpio_classic_dir=scorpio_classic_dir,
-                scorpio_dir=scorpio_dir,
-            )
-        )
-
-    gmake_opts = (
-        "{pio_dir}/Makefile -C {pio_dir} "
-        " {gmake_vars} {stdargs} -f {casetools}/Makefile".format(
-            pio_dir=pio_dir, gmake_vars=gmake_vars, casetools=casetools, stdargs=stdargs
-        )
-    )
+        cmake_opts += "-DWITH_PNETCDF:LOGICAL=FALSE -DPIO_USE_MPIIO:LOGICAL=FALSE "
+    if mpilib == "mpi-serial":
+        cmake_opts += f"-DPIO_USE_MPISERIAL=TRUE -DMPISERIAL_PATH={installpath} "
+    cmake_opts += "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON "
```

Comment:
Side question: why do we do this? I would assume that this should be added only if one issues `case.build --verbose`. I don't really care about long logs in the bldlog's...

Reply:
The buildlog for spio wouldn't have much useful info without verbose on.

Reply:
I guess this is just my personal preference. I really don't care about the bldlog details in general (and I wonder how many ppl really ever look at them _for successful builds_). I sometimes run `make VERBOSE=ON` (which has the same effect as having set `CMAKE_VERBOSE_MAKEFILE=ON` at config time), but only if I need to debug. That's why I figured "why doesn't CIME simply add VERBOSE=ON when case.build is invoked with `-v`"...

Reply:
That's a reasonable preference. I think the issue is that users have been trained for many years now to expect $package.bldlog to have the full verbose build info.